Sem. 03/Pract. 03/mySolutions/Task01.cpp

```diff
@@ -0,0 +1,166 @@
+#include <iostream>
+#include <fstream>
+
+const size_t MAX_ADDRESS_LENGTH = 26;
+const char* FILE_PATH = "orders.bin";
+
+struct PizzaOrder
+{
+	double price;
+	char address[MAX_ADDRESS_LENGTH];
+};
+
+size_t getFileSize(std::ifstream& in)
+{
+	size_t currentPos = in.tellg();
+	in.seekg(0, std::ios::end);
+	size_t fileSize = in.tellg();
+	in.seekg(currentPos);
+	return fileSize;
+}
+
+size_t strLength(const char* str)
+{
+	size_t length = 0;
+
+	while (*str != 0)
+	{
+		str++;
+		length++;
+	}
+
+	return length;
+}
+
+int strCompare(const char* str1, const char* str2)
+{
+	size_t str1Len = strLength(str1);
+	size_t str2Len = strLength(str2);
+
+	size_t lenToUse = str1Len >= str2Len ? str1Len : str2Len;
+
+	for (size_t i = 0; i < lenToUse; i++)
+	{
+		if (str1[i] > str2[i])
+			return -1;
```

Comment:
Because, if one of the words is longer than the other, the for loop will go along the length of the longer word and if it reaches the end of the shorter word and they were equal up to that point the function will return that the longer word is the bigger one (-1 or 1)

Reply:
So you compare the terminate symbol with the symbol of the longer word? This is not a clean solution because it is confusing. 
You can loop till the length of the shorter word and check the lengths after the loop with one more if. It will be more readable for others.

Reply:
Okay I understand